cmake/HalideGeneratorHelpers.cmake

```diff
@@ -608,7 +608,10 @@ function(add_halide_runtime RT)
         _Halide_fix_xcode("${RT}")
     endif ()
 
-    target_link_libraries("${RT}" INTERFACE Halide::Runtime Threads::Threads ${CMAKE_DL_LIBS})
+    target_link_libraries("${RT}" INTERFACE Halide::Runtime)
+    if (NOT (DEFINED Halide_NO_THREADS AND Halide_NO_THREADS))
+        target_link_libraries("${RT}" INTERFACE Threads::Threads ${CMAKE_DL_LIBS})
+    endif ()
     _Halide_add_targets_to_runtime("${RT}" TARGETS ${ARG_TARGETS})
```

Comment:
There should probably be something here like `if (NOT TARGET Threads::Threads)` followed by a `find_package(Threads REQUIRED)`